apps/backend/src/mutations/ExperimentMutation.ts

```diff
@@ -0,0 +1,335 @@
+      return rejection('No sample found');
+    }
+
+    const question = await this.templateDataSource.getQuestion(
+      sample.questionId
+    ); // TODO this should be a getQuestionTemplateRelation. There is no way currently of doing it. Sample should reference QuestionRel instead of Question
+    if (!question) {
+      return rejection('No question found');
+    }
+
+    const templateId = (question.config as SampleDeclarationConfig)
+      .esiTemplateId;
+    if (!templateId) {
+      return rejection('Esi template is not defined');
+    }
+
+    const newQuestionary = await this.questionaryDataSource.create(
+      user!.id,
+      templateId!
+    );
+
+    await this.questionaryDataSource.copyAnswers(
+      sample.questionaryId,
+      newQuestionary.questionaryId
+    );
+
+    return this.dataSource.addSampleToExperiment(
+      args.experimentPk,
+      args.sampleId,
+      newQuestionary.questionaryId
+    );
+  }
+
+  @Authorized()
+  async removeSampleFromExperiment(
+    user: UserWithRole | null,
+    args: RemoveSampleFromExperimentInput
+  ): Promise<ExperimentHasSample | Rejection> {
+    const experiment = await this.dataSource.getExperiment(args.experimentPk);
+    if (!experiment) {
+      return rejection('No experiment found');
+    }
+
+    const experimentSafety =
+      await this.dataSource.getExperimentSafetyByExperimentPk(
+        args.experimentPk
+      );
+
+    if (!experimentSafety) {
+      return rejection('No experiment safety found');
+    }
+
+    const hasAccessRights = await this.experimentSafetyAuth.hasWriteRights(
+      user,
+      experimentSafety.experimentSafetyPk
+    );
+    if (hasAccessRights === false) {
+      return rejection(
+        'User does not have permission to remove samples from the experimemnt',
+        {
+          args,
+        }
+      );
+    }
+
+    return this.dataSource.removeSampleFromExperiment(
+      args.experimentPk,
+      args.sampleId
+    );
+  }
+
+  @Authorized()
+  async updateExperimentSample(
+    user: UserWithRole | null,
+    args: UpdateExperimentSampleInput
+  ): Promise<ExperimentHasSample | Rejection> {
+    const experiment = await this.dataSource.getExperiment(args.experimentPk);
+    if (!experiment) {
+      return rejection('No experiment found');
+    }
+
+    const experimentSafety =
+      await this.dataSource.getExperimentSafetyByExperimentPk(
+        args.experimentPk
+      );
+
+    if (!experimentSafety) {
+      return rejection('No experiment safety found');
+    }
+
+    const hasAccessRights = await this.experimentSafetyAuth.hasWriteRights(
+      user,
+      experimentSafety.experimentSafetyPk
+    );
+    if (hasAccessRights === false) {
+      return rejection(
+        'User does not have permission to update samples in the experiment',
+        {
+          args,
+        }
+      );
+    }
+
+    return this.dataSource.updateExperimentSample(
+      args.experimentPk,
+      args.sampleId,
+      args.isSubmitted!
```

Comment:
suggestion: we should handle the undefined case as well
```suggestion
      args.isSubmitted ?? false
```